### PR TITLE
Fix mesh reader memory leak[memoryleak_and_init]

### LINKED ISSUE #328: Cubit mesh reader mismatched new / delete
I'm trying to get a valgrind clean run but got stuck when i open a mesh using the netcdf reader. There are a few mismatched new / deletes.

The Cubit mesh reader function also https://github.com/mfem/mfem/blob/master/mesh/mesh_readers.cpp#L1266 never closes the opened netCDF file.

My next idea is to just use the mesh-explorer app to circumvent the issue. I will take a deeper look on that on monday.

### PR DESCRIPTION
**Pull request by gelever**
Early return from mesh print leaks previous memory.
Reading from Netcdf leaks also.

Resolves #328.

### CODE REVIEW

**Comment by gelever** [comment]
I still need to test the netcdf portion

**Comment by tzanio** [comment]
@gelever, did you get a chance to look at this?

**Comment by tzanio** [comment]
@gelever, when you are ready: merge this branch manually in `next`, add the `in-next` tag to this PR and remove '[DON'T MERGE]' from the title :).

After a week of testing in `next`, if everything looks good, we will merge in `master`.

**Comment by gelever** [comment]
@tzanio ok, will do

**Comment by gelever** [comment]
@tzanio The netcdf mesh reader works as it should and valgrind now shows no leaks.  I don't have write access to manually merge into next, but otherwise its ready.

File: fem/intrules.hpp [hunk 1]
```diff
@@ -27,7 +27,7 @@ class IntegrationPoint
 public:
    double x, y, z, weight;
 
-   void Init() { x = y = z = weight = 0.0; }
+   IntegrationPoint() : x(0.0), y(0.0), z(0.0), weight(0.0) { }
```
Comment (v-dobrev) [inline fem/intrules.hpp]: The default constructor was intentionally left implicitly defined - that way both default-initialization and value-initialization can be used, with different behavior - similar to any fundamental type.

Is there a reason, you always want zero-initialization?

File: fem/intrules.hpp [hunk 2]
```diff
@@ -203,13 +203,7 @@ class IntegrationRule : public Array<IntegrationPoint>
...
 
    /// Construct an integration rule with given number of points
-   explicit IntegrationRule(int NP) : Array<IntegrationPoint>(NP)
-   {
-      for (int i = 0; i < this->Size(); i++)
-      {
-         (*this)[i].Init();
-      }
-   }
+   explicit IntegrationRule(int NP) : Array<IntegrationPoint>(NP) { }
```
Comment (v-dobrev) [inline fem/intrules.hpp]: The `Array` class constructor used here does not perform any initialization, that is why there was explicit initialization here. What was your reasoning behind the removal of the initialization?

File: fem/intrules.hpp (hunk 1 above)
Comment (gelever) [reply fem/intrules.hpp]: ok, i'll revert these changes

**Pull request merged**